quotefault/__init__.py

```diff
@@ -215,28 +237,138 @@ def get():
+        return redirect('/storage')
+    new_report = Report(report_id, metadata['uid'], None)
+    db.session.add(new_report)
+    db.session.commit()
+    if app.config['MAIL_SERVER'] != '':
+        send_report_email( metadata['uid'], Quote.query.get(report_id) )
+
+    flash("Report Successful!")
+    return redirect('/storage')
+
+@app.route('/review', methods=['GET'])
+@auth.oidc_auth
+def review():
+    """
+    Presents page for admins to review reports
+    and either hide or keep quote
+    """
+    metadata = get_metadata()
+    if metadata['is_admin']:
+        # Get the most recent 20 quotes
+        reports = Report.query.filter(Report.reviewed == False).all()
+
+        return render_template(
+            'bootstrap/admin.html',
+            reports=reports,
+            metadata=metadata
+        )
+    return redirect('/')
+
+@app.route('/review/<report_id>/<result>', methods=['POST'])
+@auth.oidc_auth
+def review_submit(report_id, result):
+    """
+    Called when Admin decides on a quote being hidden or kept
+    """
+    metadata = get_metadata()
+    result = int(result)
+    if metadata['is_admin']:
+        # 1 = Keep, 0 = Hide
+        if result:
+            report = Report.query.get(report_id)
+            report.reviewed = True
+            db.session.commit()
+            flash("Report Completed: Quote Kept")
+        else:
+            report = Report.query.get(report_id)
+            report.quote.hidden = True
+            report.reviewed = True
+            db.session.commit()
+            flash("Report Completed: Quote Hidden")
+        return redirect('/review')
+    return redirect('/')
+
+@app.route('/hide/<report_id>', methods=['POST'])
+@auth.oidc_auth
+def hide(report_id):
+    """
+    Hides a quote
+    """
+    metadata = get_metadata()
+    quote = Quote.query.get(report_id)
+    if ( metadata['uid'] == quote.speaker
+        or metadata['uid'] == quote.submitter
+        or metadata['is_admin'] ):
+        quote.hidden = True
+        db.session.commit()
+        flash("Quote Hidden!")
+    return redirect('/storage')
+
+@app.route('/unhide/<report_id>', methods=['POST'])
+@auth.oidc_auth
+def unhide(report_id):
+    """
+    Gives admins power to unhide a hidden quote
+    """
+    metadata = get_metadata()
+    if metadata['is_admin']:
+        quote = Quote.query.get(report_id)
+        quote.hidden = False
+        db.session.commit()
+        flash("Quote Unhidden!")
+        return redirect('/hidden')
+    return redirect('/')
+
+@app.route('/hidden', methods=['GET'])
+@auth.oidc_auth
+def hidden():
+    """
+    Presents hidden quotes to qualified users
+    Users who submitted a hidden quote or were quoted in a hidden quote will see those only
+    Admins see all hidden quotes
+    """
+    metadata = get_metadata()
+    if metadata['is_admin']:
+        quotes = Quote.query.filter( Quote.hidden ).all()
+    else:
+        quotes = Quote.query.filter(

```

Comment:
You should use the get quote query function.

Reply:
get quote query wouldn't cover the function of the hidden. It can add hidden to the query or remove them, can't show only hidden. Also can look for a specific speaker and submitter, but I would need if a specific person is either one. If anything I'd re-write a similar function below it

Reply:
get quote query returns a query, you can add filtering on top of it.

Reply:
It's meant to handle the common denominator, so we can have consistent access. If it really doesn't fit your use case, makes sense, but if it will work with an additional optional param, or as is, I'd prefer to use the helper.